### DISCUSSION TITLE
Auto backup functionality

### DISCUSSION THREAD

**Discussion by 1mfaasj** [outline/outline#6177] [Ideas]
This is my favorite wiki that I host myself and use a lot, but for that reason backups become more important. In the event of an emergency, I will have to be able to quickly retrieve all data.

I know that I can backup postgress and use the export button, but then I don't have the content of minio yet, for example, and the backup process remains manual, while I would like to automate this. I also don't want to set up a backup process in place, manually for each component (Outline, Minio and postgress). this should be much easier.

That is why I would like to see functionality in Outline that auto writes a backup file(s) at certain times/dates.
The location doesn't matter to me. the volume that docker uses for Outline for instance.

Or a functionality that makes auto backups to a (private) repo in Github.

They are just suggestions.

**Comment by tommoor** [comment]
Hi @jboesh – it is best to keep your backup solution separate from the software itself, I would recommend simply taking server disk snapshots on a regular cadence as the best way of backing up a self-hosted instance.

**Reply by 1mfaasj** [reply]
I have a snapshot service but due to its limitations (I think) I can only restore everything, not specific files/or folders like Outline's. That's why I was looking for other ways of backing up. However, I see that you can also do an export of all collections with the API. maybe I can do something with that :-)

**Reply by githubkusi** [reply]
can you also import a collection again?

**Comment by pavelcdn** [comment]
A few days ago, I encountered a serious server crash. I had to restore the wiki from a backup, but since backups are made once a day, very important information was irretrievably lost.

I really miss a built-in automatic backup feature, not just manual ones.
I tried automating it via the API, but each time I encountered insurmountable difficulties; even the AI ​​couldn't help me properly

**Reply by tommoor** [reply]
It doesn't make sense to build this into the product - there are many solutions for backing up servers already, you should replicate the postgres, for example.

**Reply by pavelcdn** [reply]
Sounds good for administrators and very advanced users, but not for housewives ))

I thought about it, but I won't be able to save files and pictures like that, In addition, you will need to think separately about where to save the database, images, and files

**Reply by githubkusi** [reply]
> It doesn't make sense to build this into the product - there are many solutions for backing up servers already, you should replicate the postgres, for example.

Doesn't need to be built in, but some recommended backup/restore scripts would be nice, such as e.g. this one here from Immich: https://docs.immich.app/administration/backup-and-restore/#restore-cli

**Comment by 1mfaasj** [comment]
I have my own backup script written in bash, if you're interested let me know

**Reply by Nordiger** [reply]
I'm interested  :-)
